includes/class-signature.php

```diff
@@ -107,8 +110,169 @@ public static function generate_signature( $user_id, $http_method, $url, $date,
 		}
 	}
 
-	public static function verify_signature( $headers, $signature ) {
+	public static function verify_http_signature( $request ) {
+		$headers = $request->get_headers();
+		$actor = isset( json_decode( $request->get_body() )->actor ) ? json_decode( $request->get_body() )->actor : '';
+		$headers['(request-target)'][0] = strtolower( $request->get_method() ) . ' /wp-json' . $request->get_route();
```

Comment:
this should be after the `if ( ! $headers )` check!

Reply:
You can rewrite the API endpoint, so it is not always `/wp-json` (see my blog for example).